numpy/ma/core.py

```diff
@@ -1032,12 +1032,7 @@ def __call__(self, a, b, *args, **kwargs):
         if m is not nomask and m.any():
             # any errors, just abort; impossible to guarantee masked values
             try:
-                np.copyto(result, 0, casting='unsafe', where=m)
-                # avoid using "*" since this may be overlaid
-                masked_da = umath.multiply(m, da)
-                # only add back if it can be cast safely
-                if np.can_cast(masked_da.dtype, result.dtype, casting='safe'):
-                    result += masked_da
```

Comment:
Yeah, that 'safe' casting is a bit weird, as nothing is done if `masked_da` cannot be converted except that the masked values are zeroed rather than set to `masked_da`. It would make more sense if an error or warning was raised on failure to convert.

Hmm, looking at it some more, I think the change here is correct.